paderwasn/rir_simulation/rir_simulation.py

```diff
@@ -0,0 +1,525 @@
+    weighting_function =\
+        calc_window(delta, kappa, toa_direct, rir_len, sample_rate)
+
+    scaling = calc_scaling(rir_ism, raw_stochastic, weighting_function,
+                           target_drr, upper_lim, lower_lim)
+    if scaling is None:
+        return None, alpha
+    else:
+        rir = rir_ism + raw_stochastic * weighting_function * scaling
+        return rir, alpha
+
+
+def calc_rirs(
+        src_pos, sensor_pos, room_dimensions, t60, azimuth_directivity,
+        colatitude_directivity, src_directivity,
+        directivity_factor_range=(2.5, 5.5), kappa=1, beta=1., rir_len=16384,
+        max_order_ism=3, win_len_direct=40, sound_speed=343, sample_rate=16000,
+        alpha_diff=0.1, max_retries=10
+):
+    """
+    Room impulse response (RIR) simulation for microphone pairs as described in
+    'DIMINISHING DOMAIN MISMATCH FOR DNN-BASED ACOUSTIC DISTANCE ESTIMATION
+    VIA STOCHASTIC ROOM REVERBERATION MODELS'
+
+    Args:
+        src_pos (array-like):
+            Coordinates of acoustic source with shape (dimension,)
+
+        sensor_pos (array-like):
+            Coordinates of the two microphones of the microphone pair with
+            shape (dimension, 2). It is assumed that the microphones have a
+            reasonably spacing for the assumnptions to hold.
+
+        room_dimensions (array-like):
+            Size of the room with shape (dimension)
+
+        t60 (float):
+            T60 time of the room in seconds
+
+        azimuth_directivity (float):
+            Azimuth of source in degrees
+
+        colatitude_directivity (float):
+            Colatitude of source in degrees
+
+        src_directivity (str):
+            Directivity pattern of the source; possible choices:
+            'cardioid', 'hypercardioid', 'supercardioid' 'subcardioid' and
+            'omnidirectional'
+
+        directivity_factor_range (array-like):
+            Range from which the directional factor of the sound source (alpha)
+            is drawn uniformly
+
+        beta (float):
+            Directivity factor of microphone
+
+        kappa (float):
+            Determines the fade-in of the stochastic part of the RIR.
+            Lower values cause smoother transition.
+
+        rir_len (int):
+            Length of RIR to be generated in samples
+
+        max_order_ism (int):
+            Order of simulated image sources
+
+        win_len_direct (int):
+            One-sided length of the window defining the part of the RIR
+            which belongs to the direect path.
+
+        sound_speed (int):
+            Speed of sound used to determine the position of the direct path
+
+        sample_rate (int):
+            Sample rate
+
+        alpha_diff (float):
+            Maximum difference between alpha factors used for RIRs of different
+            microphones to be tolerated.
+
+        max_retries (int):
+            Number of iterations to re-draw alpha factor if no solution could
```

Comment:
Number of iterations to re-draw alpha factor; If no valid solution for the scaling of the stochastic part of the RIR  could be found to maintain the target DRR  before the maximum value of iteration is reached, the interval from which alpha is drawn is adapted to the setup at hand.